assets/src/components/activities/adaptive/AdaptiveDelivery.tsx

```diff
@@ -192,16 +192,29 @@ const Adaptive = (props: DeliveryElementProps<AdaptiveModelSchema>) => {
     [parts, adaptivityDomain],
   );
 
-  const handlePartInit = async (payload: { id: string | number; responses: any[] }) => {
-    /* console.log('onPartInit', payload); */
+  const handlePartInit = async (payload: {
+    id: string | number;
+    responses: any[];
+    expressions?: string[];
+  }) => {
     // a part should send initial state values
     if (payload.responses.length) {
       const saveResults = await handlePartSave(payload);
     }
-
+    let expressionResult: any = [];
+    if (payload.expressions && payload.expressions.length && props.onInitPart) {
```

Comment:
seems like if we're going to have a part init event then we might as well fire it all the time, and the result of it should be a payload object (that could possibly contain "expressions" in the result). this would make it more generic for all activities to know what to expect.